src/org/opensolaris/opengrok/analysis/vb/Consts.java

```diff
@@ -19,54 +19,193 @@
 
 /*
  * Copyright (c) 2007, 2013, Oracle and/or its affiliates. All rights reserved.
+ * Portions Copyright (c) 2017, Chris Fraire <cfraire@me.com>.
  */
 package org.opensolaris.opengrok.analysis.vb;
 
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStreamReader;
 import java.util.Collections;
 import java.util.HashSet;
-import java.util.Locale;
 import java.util.Set;
 
 @SuppressWarnings("PMD.AvoidThrowingRawExceptionTypes")
 public final class Consts {
 
-    private static final Set<String> reservedKeywords;
+    public static final Set<String> reservedKeywords;
+    public static final Set<String> directives;
 
     static {
-        HashSet<String> kwds = new HashSet<String>();
-        try {
-
-            populateKeywordSet(kwds, "vbReservedWords.dat");
-        } catch (IOException ioe) {
-            throw new RuntimeException(ioe);
-        }
+        HashSet<String> kwds = new HashSet<>();
+        populateKeywordSet(kwds);
         reservedKeywords = Collections.unmodifiableSet(kwds);
+
+        HashSet<String> hashwords = new HashSet<>();
+        hashwords.add("#const");	// VB lang-reference/keywords
+        hashwords.add("#else");	// VB lang-reference/keywords
+        hashwords.add("#elseif");	// VB lang-reference/keywords
+        hashwords.add("#end");	// VB lang-reference/keywords
+        hashwords.add("#if");	// VB lang-reference/keywords
+        directives = Collections.unmodifiableSet(hashwords);
     }
 
     private Consts() {
         // Util class, can not be constructed.
     }
 
-    private static void populateKeywordSet(Set<String> set, String file)
-            throws IOException {
-        String line, lline;
-        try (BufferedReader reader = new BufferedReader(new InputStreamReader(
-                    Consts.class.getResourceAsStream(file), "US-ASCII"))) {
-            while ((line = reader.readLine()) != null) {
-                line = line.trim();
-                lline = line.toLowerCase(Locale.US);
-                if (line.charAt(0) != '#') {
-                    set.add(line);
-                    set.add(lline);
-                }
-            }
-        }
-    }
+    private static void populateKeywordSet(Set<String> kwd) {
```

Comment:
we're loosing precision here ... so this is not a showstopper, but I am just thinking aloud ...
I think we will want to have case sensitivity soon, new lucene has no problems with saving the index in case sensitive format and query it insensitively now ... (with 7.x this will be even more obvious)
now we will loose some of the case sensitivity info ...

Reply:
This patch does not change the values that get stored as symbols — see the unchanged `setAttribs()` call. It changes the lexer to be strictly case-insensitive to accord with the language and changes the `Consts` to reflect this. VB symbols that were actually keywords but not previously found in the case-sensitive set are now properly identified as keywords.

(But yes, if the actual search needs to be case-insensitive, then I've seen the existing ideas on this from the historical Issues list.)

Reply:
ok, so my concerns are not valid, this is good, so if VB spec expects lowercase, then this fix is perfect, thank you